arch/riscv/core/pmp.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2022 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <linker/linker-defs.h>
+#include <pmp.h>
+#include <sys/arch_interface.h>
+#include <arch/riscv/csr.h>
+
+#define LOG_LEVEL CONFIG_MPU_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mpu);
+
+#define PMP_DEBUG_DUMP 0
```

Comment:
Tried setting this to one, but printing hangs (tried the userspace hello_world) on qemu...

Reply:
Works fine here. Did you modify something else?

Reply:
I can confirm that it works correctly on my setup with `PMP_DEBUG_DUMP` set to `1`. 

On another note, I think that this setting should be exposed somehow, e.g. by an option in Kconfig.

Reply:
Well... the actual PMP updates are already logged when the PMU log level is set to DEBUG in Kconfig. Dumping the whole register content is useful only if the PMP code itself is bad (as opposed to memory comain config issues) and you must be ready to modify that code anyway. So I'd leave that one hidden from casual Kconfig alterations.

Reply:
Not that I remember, but after getting last version, it seems to work. I'll blame my env then...